lib/rules/no-invalid-role.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const AstNodeInfo = require('../helpers/ast-node-info');
+const Rule = require('./base');
+
+function ERROR_MESSAGE_INVALID_ROLE(element) {
+  return `Use of presentation role on <${element}> detected. Semantic elements should not be used for presentation.`;
+}
+
+// None of these elements can be marked with `role="presentation"` or `role="none"`.
+// List from https://developer.mozilla.org/en-US/docs/Web/HTML/Element.
+const DISALLOWED_ELEMENTS = [
+  'a',
+  'abbr',
+  'applet',
+  'area',
+  'audio',
+  'b',
+  'bdi',
+  'bdo',
+  'blockquote',
+  'br',
+  'button',
+  'caption',
+  'cite',
+  'code',
+  'col',
+  'colgroup',
+  'data',
+  'datalist',
+  'dd',
+  'del',
+  'details',
+  'dfn',
+  'dialog',
+  'dir',
+  'dl',
+  'dt',
+  'em',
+  'embed',
+  'fieldset',
+  'figcaption',
+  'figure',
+  'form',
+  'hr',
+  'i',
+  'iframe',
+  'input',
+  'ins',
+  'kbd',
+  'label',
+  'legend',
+  'li',
+  'main',
+  'map',
+  'mark',
+  'menu',
+  'menuitem',
+  'meter',
+  'noembed',
+  'object',
+  'ol',
+  'optgroup',
+  'option',
+  'output',
+  'p',
+  'param',
+  'pre',
+  'progress',
+  'q',
+  'rb',
+  'rp',
+  'rt',
+  'rtc',
+  'ruby',
+  's',
+  'samp',
+  'select',
+  'small',
+  'source',
+  'strong',
+  'sub',
+  'summary',
+  'sup',
+  'table',
+  'tbody',
+  'td',
+  'textarea',
+  'tfoot',
+  'th',
+  'thead',
+  'time',
+  'tr',
+  'track',
+  'tt',
+  'u',
+  'ul',
+  'var',
+  'video',
+  'wbr',
+];
+
+module.exports = class NoInvalidRole extends Rule {
+  visitor() {
+    return {
+      ElementNode(node) {
+        const hasRoleAttribute = AstNodeInfo.hasAttribute(node, 'role');
+        const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
+        if (hasRoleAttribute) {
+          if (['presentation', 'none'].includes(roleValue)) {
+            if (DISALLOWED_ELEMENTS.includes(node.tag)) {
```

Comment:
looks like a lot of nesting here. do you think it might be worth it to collapse that into a single condition with `&&`? or do something like `if (!hasRoleAttribute) return;`?

Reply:
Previous reviews have requested that I don't use `&&` but rather nest it for easier maintainability - I'll see if I can simplify though.

Reply:
🤔 

seems like an interesting conversation. not sure I'd support that opinion... 😅 

in this case I'd personally go with the early-return pattern, because I assume that more `role` related assertions will follow